sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ParallelDocumentQueryTest.java

```diff
@@ -265,6 +265,79 @@ public void queryDocumentsWithCompositeContinuationTokens() throws Exception {
+        options.setEnableCrossPartitionQuery(true);
+        options.setMaxDegreeOfParallelism(2);
+
+        List<String> expectedValues = createdDocuments.stream().map(d -> d.getId()).collect(Collectors.toList());
+
+        String query = "Select value c.id from c";
+
+        Flux<FeedResponse<String>> queryObservable = createdCollection.queryItems(query, options, String.class);
+
+        List<String> fetchedResults = new ArrayList<>();
+        queryObservable.map(stringFeedResponse -> fetchedResults.addAll(stringFeedResponse.getResults())).blockLast();
+
+        assertThat(fetchedResults).containsAll(expectedValues);
+    }
+
+    @Test(groups = { "simple" })
+    public void queryDocumentsArrayValue(){
+        FeedOptions options = new FeedOptions();
+        options.setEnableCrossPartitionQuery(true);
+        options.setMaxDegreeOfParallelism(2);
+
+        List<List<Integer>> expectedValues = new ArrayList<>();
+        ArrayList<Integer> a1 = new ArrayList<Integer>();
+        ArrayList<Integer> a2 = new ArrayList<Integer>();
+        a1.add(6519456);
+        a1.add(1471916863);
+        a2.add(2498434);
+        a2.add(1455671440);
+        expectedValues.add(a1);
+        expectedValues.add(a2);
+
+        String query = "Select top 2 value c.sgmts from c";
+
+        Flux<FeedResponse<List>> queryObservable = createdCollection.queryItems(query, options, List.class);
+
+        List<List> fetchedResults = new ArrayList<>();
+        queryObservable.map(feedResponse -> fetchedResults.addAll(feedResponse.getResults())).blockLast();
+    }
+
+    @Test(groups = { "simple" })
+    public void queryDocumentsIntegerValue(){
+        FeedOptions options = new FeedOptions();
+        options.setEnableCrossPartitionQuery(true);
+        options.setMaxDegreeOfParallelism(2);
+
+        List<Integer> expectedValues = createdDocuments.stream().map(d -> d.getInt("prop")).collect(Collectors.toList());
+
+        String query = "Select value c.prop from c";
+
+        Flux<FeedResponse<Integer>> queryObservable = createdCollection.queryItems(query, options, Integer.class);
+
+        List<Integer> fetchedResults = new ArrayList<>();
+        queryObservable.map(feedResponse -> fetchedResults.addAll(feedResponse.getResults())).blockLast();
+
+        assertThat(fetchedResults).containsAll(expectedValues);
+    }
+
+    @Test(groups = { "simple" })
+    public void queryDocumentsPojo(){
+        FeedOptions options = new FeedOptions();
+        options.setEnableCrossPartitionQuery(true);
+        options.setMaxDegreeOfParallelism(2);
+        String query = "Select * from c";
+        Flux<FeedResponse<TestObject>> queryObservable = createdCollection.queryItems(query, options, TestObject.class);
+        List<TestObject> fetchedResults = new ArrayList<>();
+        queryObservable.map(feedResponse -> fetchedResults.addAll(feedResponse.getResults())).blockLast();
+        System.out.println("fetchedResults = " + fetchedResults);
```

Comment:
I am not seeing any assertion in the test. Does test verify that the result is correct?

Reply:
Added assertions